### PR TITLE
fix: skip None output instead of throw an error

### PR DESCRIPTION
**Pull request by void-main**
# Description
For some models, there may be `None` output for scripted model, for example, in `torchvision.inception_v3`, the second output is a None constant.

Current implementation throws error as reported in issue #469 .

I added a special case for `out_ivalue.isNone()` and skip the output to make the model compile.

## Call for discussion
Personally I don't know if it's a good idea to skip an output, it may cause error in client code (out of index error when try to get the output), but I don't know how to build a `None` tensor with TensorRT.

Fixes # (issue)
#469 

## Type of change

Please delete options that are not relevant and/or add your own.

- Bug fix (non-breaking change which fixes an issue)

# Checklist:

- [x] My code follows the style guidelines of this project (You can use the linters)
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas and hacks
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests to verify my fix or my feature
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: core/conversion/conversion.cpp [hunk 1]
```diff
@@ -229,6 +229,11 @@ void MarkOutputs(ConversionCtx* ctx, at::ArrayRef<const torch::jit::Value*> outp
           LOG_INFO(
               ctx->logger, "Marking Output " << out->debugName() << " named " << name << " in engine (ctx.MarkOutput)");
           ctx->num_outputs += 1;
+        } else if (out_ivalue.isNone()) { // skip a None output layer
+          std::string name = std::string("output_") + std::to_string(ctx->num_outputs);
+          LOG_INFO(
```
Comment (peri044) [inline core/conversion/conversion.cpp:234]: can you make it LOG_WARNING instead ?

**Review by peri044 — APPROVED** [review]

**Comment by peri044** [comment]
@narendasan  the fix seems ok to me. 
The graph for inceptionv3 scripted has the following 
```
%self.Conv2d_1a_3x3.conv.bias.1 : NoneType = prim::Constant()
%1611 : str = prim::Constant[value="Scripted Inception3 always returns Inception3 Tuple"]()
....
....
return (%1955, %self.Conv2d_1a_3x3.conv.bias.1)
```

Maybe this is one weird case of scripting inception models ?. This fix seems reasonable to me.

File: core/conversion/conversion.cpp (hunk 1 above)
Comment (void-main) [reply core/conversion/conversion.cpp:234]: Sure, I'll do a quick patch.

**Comment by void-main** [comment]
> @narendasan the fix seems ok to me. The graph for inceptionv3 scripted has the following
> 
> ```
> %self.Conv2d_1a_3x3.conv.bias.1 : NoneType = prim::Constant()
> %1611 : str = prim::Constant[value="Scripted Inception3 always returns Inception3 Tuple"]()
> ....
> ....
> return (%1955, %self.Conv2d_1a_3x3.conv.bias.1)
> ```
> 
> Maybe this is one weird case of scripting inception models ?. This fix seems reasonable to me.

yep, the scripted model looks pretty weird, do you have any idea on how to create a `NoneType` tensor?

**Comment by peri044** [comment]
@void-main  @narendasan Looks like this is a special case where they are using `aux` for training which is by default optional null tensor 
https://github.com/pytorch/vision/blob/main/torchvision/models/inception.py#L135-L155

**Comment by narendasan** [comment]
I think what we really need to make sure is that the graph construction phase doesnt break because the input indexes dont meet expectations. For example what happens if the tensor in the middle is None instead of one on the end. Would producing a dummy output be a better option here vs skipping? Maybe we can explicitly produce a NoneTensor in the graph?

**Comment by void-main** [comment]
@narendasan totally agree that we should produce a NoneTensor in the graph. Could you please point out how to create a NoneTensor? I'll change my code.

**Comment by void-main** [comment]
Got it, I'll try to make these changes later this week.

**Pull request closed without merging**